yb-voyager/cmd/importData.go

```diff
@@ -1168,14 +1168,31 @@ func extractCopyStmtForTable(table string, fileToSearchIn string) {
 		return
 	}
 	// pg_dump and ora2pg always have columns - "COPY table (col1, col2) FROM STDIN"
-	copyCommandRegex := regexp.MustCompile(fmt.Sprintf(`(?i)COPY %s[\s]+\(.*\) FROM STDIN`, table))
+	var copyCommandRegex *regexp.Regexp
 	if sourceDBType == "postgresql" {
 		// find the line from toc.txt file
 		fileToSearchIn = exportDir + "/data/toc.txt"
 
-		// if no schema then add public in tableName as it is there in postgres' toc file
-		if len(strings.Split(table, ".")) == 1 {
-			copyCommandRegex = regexp.MustCompile(fmt.Sprintf(`(?i)COPY public.%s[\s]+\(.*\) FROM STDIN`, table))
+		conn := newTargetConn()
+		defer conn.Close(context.Background())
+
+		parentTable := getParentTable(table, conn)
+		if parentTable != "" { // this is partitioned table
+			// add schema name to parentTable
+			splits := strings.Split(table, ".")
+			if len(splits) > 1 {
+				parentTable = splits[0] + "." + parentTable
+			} else {
+				parentTable = "public." + parentTable
+			}
+			copyCommandRegex = regexp.MustCompile(fmt.Sprintf(`(?i)COPY %s[\s]+\(.*\) FROM STDIN`, parentTable))
+		} else {
+			// if no schema then add public in tableName as it is there in postgres' toc file
```

Comment:
This comment probably does not belong here..

Reply:
why? It specific to the next line